### PR TITLE
feat: add string type for text variables

### PR DESCRIPTION
**Pull request by vorel99**
It would be nice to have specialt type for plain text variables like name, address etc.
I created string type, that represents those types of variables. For now, I took most of the functionality from categorical variable, but I think, that in time description and render can be pretty different.
- added new data type String for plain text variables
- added describe function for string
- added render function for string
  - render word cloud instead of contingency table

### CODE REVIEW

**Comment by fabclmnt** [comment]
Hi @vorel99,

thank you for your contribution! Seems like an interesting context and indeed something that can benefit the community!

File: src/ydata_profiling/model/summary_algorithms.py [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
 import numpy as np
-import pandas as pd
 from multimethod import multimethod
 from scipy.stats import chisquare
-
 from ydata_profiling.config import Settings
 
+import pandas as pd
+
```
Comment (alexbarros) [inline src/ydata_profiling/model/summary_algorithms.py:10]: why the change in the import order?

File: src/ydata_profiling/model/typeset_relations.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def inner(series: pd.Series, *args, **kwargs) -> bool:
     return inner
 
 
-def string_is_bool(series: pd.Series, state: dict, k: Settings) -> bool:
+def string_is_bool(series: pd.Series, state: dict, k: Dict[str, bool]) -> bool:
```
Comment (alexbarros) [inline src/ydata_profiling/model/typeset_relations.py:27]: why the signature change?

File: src/ydata_profiling/model/spark/describe_string_spark.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    Args:
+        series: The Series to describe.
+        summary: The dict containing the series description so far.
+
+    Returns:
+        A dict containing calculated series description values.
+    """
+
+    # FIXME: cat description
```
Comment (alexbarros) [inline src/ydata_profiling/model/spark/describe_string_spark.py:22]: fix the fixmme or remove it plz

File: src/ydata_profiling/visualisation/plot.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 from matplotlib.ticker import FuncFormatter
 from statsmodels.graphics.tsaplots import plot_acf, plot_pacf
 from typeguard import typechecked
-
+from wordcloud import WordCloud
```
Comment (alexbarros) [inline src/ydata_profiling/visualisation/plot.py:16]: I believe this was not included in the requirements

File: src/ydata_profiling/visualisation/plot.py [hunk 5]
```diff
@@ -24,6 +24,23 @@ def format_fn(tick_val: int, tick_pos: Any) -> str:
...
+def _plot_word_cloud(
+    series: pd.Series,
+    figsize: tuple = (6, 4),
+) -> plt.Figure:
+    word_dict = series.to_dict()
+    wordcloud = WordCloud(
+        background_color="white",
+        random_state=123,
+        width=500,
+        height=300,
```
Comment (alexbarros) [inline src/ydata_profiling/visualisation/plot.py:36]: ideally, the size should be in the configuration

File: tests/unit/test_typeset_custom.py [hunk 6]
```diff
@@ -147,23 +157,12 @@ def get_profiling_series():
...
-        self.name = name
-        self.contains_type = contains_type
-        self.infer_type = infer_type
-        self.cast_result = cast_result
-
-
-cases = [
-    DataTest("x", Numeric, Numeric),
-]
-
```
Comment (alexbarros) [inline tests/unit/test_typeset_custom.py:166]: was this not used?

File: src/ydata_profiling/visualisation/plot.py [hunk 7]
```diff
@@ -99,6 +116,17 @@ def _plot_histogram(
...
 
 
+@manage_matplotlib_context()
+def plot_word_cloud(
+    config: Settings, word_counts: pd.Series, mini: bool = False
+) -> str:
+    if mini:
+        plot = _plot_word_cloud(series=word_counts, figsize=(3, 2.25))
+    else:
+        plot = _plot_word_cloud(series=word_counts)
```
Comment (alexbarros) [inline src/ydata_profiling/visualisation/plot.py:126]: `plot` is not used

**Review by alexbarros — CHANGES_REQUESTED** [review]
I believe the new type is a good idea, but there are some critical issues that need to be solved first:
- add `wordcloud` to requirements
- adjust the renders class to match the other renders
- validate that the tests are running correctly
- pass all the automated checks

It would also be interesting to add a few screenshots of the new variable profile.

File: src/ydata_profiling/model/summary_algorithms.py (hunk 1 above)
Comment (vorel99) [reply src/ydata_profiling/model/summary_algorithms.py:10]: I'm using automatic import sorter. I could change that for this MR

**Comment by vorel99** [comment]
Hello,
Thank you for your reply.
I'm sorry for poor quality of my merge request, it's my first merge request ever to community repo.
Tomorrow I will resolve noted issues.

File: src/ydata_profiling/report/structure/variables/render_string.py [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+    _get_n,
+    freq_table,
+    render_categorical_frequency,
+    render_categorical_length,
+    render_categorical_unicode,
+)
+from ydata_profiling.visualisation.plot import plot_word_cloud
+
+
+class RenderString(BaseRenderVariable):
```
[the remark this answers is not in the document]
Comment (vorel99) [reply src/ydata_profiling/report/structure/variables/render_string.py:22]: I could change other renders to use same BaseRenderVariable, but I presume, that should be in separate MR. Am I right?

File: src/ydata_profiling/model/typeset_relations.py (hunk 2 above)
Comment (vorel99) [reply src/ydata_profiling/model/typeset_relations.py:27]: becouse as `k`, we are sending just `Setting.vars.bool.mappings`, not whole `Setting`

**Comment by vorel99** [comment]
Here is demo of string data type.

Now it has same information as categorical type, but I think, there are some information in categorical type, that should be just in string type. For example `words` and `characters` tabs. 
I didn't delete them yet, I would prefer your point of view. 

string_top
string_bottom

File: tests/unit/test_typeset_custom.py (hunk 6 above)
Comment (vorel99) [reply tests/unit/test_typeset_custom.py:166]: yes, it was unused

**Comment by alexbarros** [comment]
>... I'm sorry for poor quality of my merge request, it's my first merge request ever to community repo.

No worries, your pull request is actually pretty good, most of the issues are related to some conventions that we need to observe to keep consistent the contributions from the community. In this sense, the commit messages in this PR need to comply with the https://www.conventionalcommits.org/en/v1.0.0/, since we use it to automate some of the GitHub actions.

Also ,you may need to run the `pre-commit` to avoid failing the linter checks
```
$ pre-commit install
$ pre-commit run --all-files
```
ps. pre-commit is in the dev requirements

File: tests/unit/test_typeset_default.py [hunk 9]
```diff
@@ -177,7 +180,7 @@ def test_contains(name, series, contains_type, member):
 
 inference_map = {
     "int_series": Numeric,
-    "categorical_int_series": Numeric,
+    "categorical_int_series": Categorical,
```
Comment (alexbarros) [inline tests/unit/test_typeset_default.py:183]: this and the float one are a bit odd, I didn't see any changes that should affect this test. It's not an issue but odd nonetheless.

File: src/ydata_profiling/model/pandas/describe_text_pandas.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        A dict containing calculated series description values.
+    """
+
+    series = series.astype(str)
+
+    # Only run if at least 1 non-missing value
+    value_counts = summary["value_counts_without_nan"]
+    value_counts.index = value_counts.index.astype(str)
+
+    summary.update({"first_rows": series.head(5)})
```
Comment (alexbarros) [inline src/ydata_profiling/model/pandas/describe_text_pandas.py:44]: ignoring redact here, minor issue since redact will be accessed in the render

**Review by alexbarros — COMMENTED** [review]
Codewise looks good to me. But now we need to solve the version compatibility, it seems that `wordcloud` officially supports python 3.6 and 3.7 while we support 3.6-3.11. Will check if the failed CI flow is related to version incompatibility.

**Comment by alexbarros** [comment]
I checked their wordcloud repo and it seems that they have some commits related to support to python 3.11, I posted an issue related to when there will be a new release, but without it either we find an alternative for the package, replace the wordcloud for another visualization or put the PR on hold

**Comment by fabclmnt** [comment]
Hi @vorel99, 

thank you for the contribution. As mentioned by @alexbarros we will be putting this PR on-hold unless we find a package that is more suited to deliver the word of cloud experience.

**Review by alexbarros — APPROVED** [review]

**Comment by alexbarros** [comment]
@vorel99 since the new release of the newest version of wordcloud the issue with the python version support was fixed, so we are moving on to merge this.
Thanks for your contribution!
